docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,666 @@
+   - [Connecting different sub-types of the same super-type, with colliding names](#connecting-different-sub-types-of-the-same-super-type-with-colliding-names)
+   - [Connecting sub-types to super-types by waiving extra members](#connecting-sub-types-to-super-types-by-waiving-extra-members)
+   - [Connecting different sub-types](#connecting-different-sub-types)
+
+## Terminology
+
+- "Chisel type" - a `Data` that is not bound to hardware, i.e. not a component. (more details [here](chisel-type-vs-scala-type)).
+  - E.g. `UInt(3.W)`, `new Bundle {..}`, `Vec(3, SInt(2.W))` are all Chisel types
+- `Aggregate` - a Chisel type or component that contains other Chisel types or components (i.e. `Vec`, `Record`, or `Bundle`)
+- `Element` - a Chisel type or component that does not contain other Chisel types or components (e.g. `UInt`, `SInt`, `Clock`, `Bool` etc.)
+- "component" - a `Data` that is bound to hardware (`IO`, `Reg`, `Wire`, etc.)
+  - E.g. `Wire(UInt(3.W))` is a component, whose Chisel type is `UInt(3.W)`
+- "member" - a Chisel type or component, or any of its children (could be an `Aggregate` or an `Element`)
+  - E.g. `Vec(3, UInt(2.W))(0)` is a member of the parent `Vec` Chisel type
+  - E.g. `Wire(Vec(3, UInt(2.W)))(0)` is a member of the parent `Wire` component
+  - E.g. `IO(Decoupled(Bool)).ready` is a member of the parent `IO` component
+- "relative alignment" - whether two members of the same component or Chisel type are aligned/flipped, relative to one another
+  - see section [below](#alignment-flipped-vs-aligned) for a detailed definition
+- "structural type check" - Chisel type `A` is structurally equivalent to Chisel type `B` if `A` and `B` have matching bundle field names and types (`Record` vs `Vector` vs `Element`), vector sizes, `Element` types (UInt/SInt/Bool/Clock etc))
+  - ignores relative alignment (flippedness)
+- "alignment type check" - a Chisel type `A` matches alignment with another Chisel type `B` if every member of `A`'s relative alignment to `A` is the same as the structurally corresponding member of `B`'s relative alignment to `B`.
+
+## Overview
+
+The `Connectable` operators are the standard way to connect Chisel hardware components to one another.
+
+> Note: For descriptions of the semantics for the previous operators, see [`Connection Operators`](connection-operators).
+
+All connection operators require the two hardware components (consumer and producer) to be structurally type equivalent.
+
+The one exception to the structural type-equivalence rule is using the `Connectable` mechanism, detailed at this [section](#waived-data) towards the end of this document.
+
+Aggregate (`Record`, `Vec`, `Bundle`) Chisel types can include data members which are flipped relative to one another. Due to this, there are many desired connection behaviors between two Chisel components. The following are the Chisel connection operators:
```

Comment:
```suggestion
Aggregate (`Record`, `Vec`, `Bundle`) Chisel types can include data members which are flipped relative to one another.
Due to this, there are many desired connection behaviors between two Chisel components.
The following are the Chisel connection operators:
```

Because newlines don't impact the result, it's generally advisable in markdown (and really most markdown/up formats) to include newlines between sentences which reduces diff noise/merge conflicts when editing the files in the future.